### PR TITLE
trio v0.16.0

### PR DESCRIPTION
**Pull request by thopkins32**
It is very likely that the current package version for this feedstock is out of date.
Notes for merging this PR:
1. Feel free to push to the bot's branch to update this PR if needed.
2. The bot will almost always only open one PR per version.
Checklist before merging this PR:
- [ ] Dependencies have been updated if changed: see [upstream](https://github.com/python-trio/trio)
- [ ] Tests have passed 
- [ ] Updated license if changed and `license_file` is packaged 

Note that the bot will stop issuing PRs if more than 3 Version bump PRs generated by the bot are open. If you don't want to package a particular version please close the PR.

**NEW: If you want these PRs to be merged automatically, make an issue with <code>@conda-forge-admin,</code>`please add bot automerge` in the title and merge the resulting PR. This command will add our new bot automerge feature to your feedstock!**

If this PR was opened in error or needs to be updated please add the `bot-rerun` label to this PR. The bot will close this PR and schedule another one. If you do not have permissions to add this label, you can use the phrase <code>@<space/>conda-forge-admin, please rerun bot</code> in a PR comment to have the `conda-forge-admin` add it for you.

<sub>This PR was created by the [regro-cf-autotick-bot](https://github.com/regro/cf-scripts).
The **regro-cf-autotick-bot** is a service to automatically track the dependency graph, migrate packages, and propose package version updates for conda-forge. If you would like a local version of this bot, you might consider using [rever](https://regro.github.io/rever-docs/). Rever is a tool for automating software releases and forms the backbone of the bot's conda-forge PRing capability. Rever is both conda (`conda install -c conda-forge rever`) and pip (`pip install re-ver`) installable.
Finally, feel free to drop us a line if there are any [issues](https://github.com/regro/cf-scripts/issues)!
This PR was generated by , please use this URL for debugging</sub>

### CODE REVIEW

**Comment by thopkins32** [comment]
@mrakitin This was the bot running on my account for updating the version of `trio` to `v0.16.0`. It looks like they have their own build infrastructure that they use compared to ours. Such as using their own docker image, forcing changes to their style in the `meta.yaml` and other configurations for Azure. Let me know what changes I should make to get it to something we want instead.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
+{% set version = "0.16.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  {{ hash_type }}: {{ hash_value }}
+  sha256: df067dd0560c321af39d412cd81fc3a7d13f55af9150527daab980683e9fcf3c
```
Comment (mrakitin) [inline recipe/meta.yaml:10]: I'd like to keep the hash parametrized on top of the file, if possible. That way bumping a version number appears on one chunk.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 
 build:
   number: 0
-  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
-  skip: True  # [py<36]
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
+  skip: true  # [py<36]
```
Comment (mrakitin) [inline recipe/meta.yaml:15]: This is a good change, I like it.

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 3]
```diff
@@ -6,19 +6,29 @@ jobs:
...
-        UPLOAD_PACKAGES: True
+      linux_64_python3.6.____cpython:
+        CONFIG: linux_64_python3.6.____cpython
+        UPLOAD_PACKAGES: 'True'
         DOCKER_IMAGE: condaforge/linux-anvil-comp7
-      linux_python3.7:
-        CONFIG: linux_python3.7
-        UPLOAD_PACKAGES: True
+      linux_64_python3.7.____cpython:
+        CONFIG: linux_64_python3.7.____cpython
```
Comment (mrakitin) [inline .azure-pipelines/azure-pipelines-linux.yml]: This `*___cpython` caused a few problems for some combination of platforms/python versions. I now specify python versions explicitly in `conda_build_config.yaml` file. Probably worth to rerender here after the change is done.

**Review by mrakitin — COMMENTED** [review]
Looks nice! Did the bot script rerender it automatically, or you did it manually?

**Comment by thopkins32** [comment]
The bot rerenders the feedstock automatically.

File: recipe/meta.yaml (hunk 1 above)
Comment (thopkins32) [reply recipe/meta.yaml:10]: I'll look into where this change takes place today.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 3 above)
Comment (thopkins32) [reply .azure-pipelines/azure-pipelines-linux.yml]: There currently is no python version specified in the `conda_build_config.yaml` at the moment. Do you want me to manually make the change and rerender?

Do we need to change this back to the original two lines?

File: recipe/meta.yaml (hunk 1 above)
Comment (mrakitin) [reply recipe/meta.yaml:10]: Thanks!

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 3 above)
Comment (mrakitin) [reply .azure-pipelines/azure-pipelines-linux.yml]: Yes please. You can use the file from the `event-model` feedstock, but please add more python versions to the list as this package is arch-dependent.
https://github.com/nsls-ii-forge/event-model-feedstock/blob/5d88d0dce9464dcd9c7a91c54eecdd390638d943/recipe/conda_build_config.yaml#L10

Comment (mrakitin) [reply .azure-pipelines/azure-pipelines-linux.yml]: Manual rerendering is needed then.

Comment (thopkins32) [reply .azure-pipelines/azure-pipelines-linux.yml]: I can change the rerender command to use the `conda_build_config.yaml` if it exists. Should be a simple change.

File: recipe/meta.yaml (hunk 1 above)
Comment (thopkins32) [reply recipe/meta.yaml:10]: So I found where it makes this change [here](https://github.com/regro/cf-scripts/blob/3a757fcc27806e882f497e3b79ce7f77b9054805/conda_forge_tick/migrators/extra_jinj2a_keys_cleanup.py#L12). It says that it makes this change to make it easier for the version migrator to do its job. I can try excluding the `ExtraJinja2KeysCleanup` migrator and see what happens if you want.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 3 above)
Comment (mrakitin) [reply .azure-pipelines/azure-pipelines-linux.yml]: It should not be an exclusive config. Regular rerender command should be OK. I think we just need to manually update the file, rerender and rerun the bot.

By the way, does the bot keep track of the already opened PRs with the same update?

File: recipe/meta.yaml (hunk 1 above)
Comment (mrakitin) [reply recipe/meta.yaml:10]: OK, sounds good, it's time to experiment!

**Pull request closed without merging**